specs/lib/addPipelineExtrasSpec.js

```diff
@@ -6,7 +6,6 @@ var WebGLConstants = Cesium.WebGLConstants;
 
 describe('addPipelineExtras', function() {
     it('adds pipeline extras', function() {
-        // TODO: KHR_techniques_webgl - remove shaders from top level, put inside extras.KHR_techniques_webgl
```

Comment:
Looking at this I realized we will have to fix `ForEach.shaderLegacy` to work for both 1.0 and 2.0 models, otherwise `addPipelineExtras` will not work if a 2.0 + `KHR_techniques_webgl` model comes through.

Which makes me think that to avoid this sort of confusion we should actually just merge the legacy and non-legacy functions. Every function in `ForEach` should be able to work for 1.0 or 2.0, and helpers like `ForEach.topLevel` and `ForEach.object` should work for both as well. Hopefully not too many of the functions need to change as long as the helpers are updated.

Sorry for the sudden change, but I think it will be simpler that way. I was never super happy with the legacy vs. non-legacy approach.

For example `ForEach.shader` might look like:
```js
ForEach.shader = function(gltf, handler) {
    if (hasExtension(gltf, 'KHR_techniques_webgl')) {
        ForEach.object(gltf.extensions.KHR_techniques_webgl.shaders, handler);
    else if (defined(gltf.shaders)) {
        ForEach.topLevel(gltf, 'shaders', handler);
    }
};
```

Once those changes are done the test can be fixed to use gltf 2.0 + `KHR_techniques_webgl` syntax rather than the current 1.0 syntax.